hypex/comparators/abstract.py

```diff
@@ -125,41 +138,164 @@ def _extract_dataset(
         return Dataset.from_dict(compare_result, roles, BackendsEnum.pandas)
 
     # TODO выделить в отдельную функцию с кваргами (нужно для альфы)
+
+    @staticmethod
+    def _split_ds_into_columns(
+        data: List[Tuple[str, Dataset]]
+    ) -> List[Tuple[str, Dataset]]:
+        result = []
+        for bucket in data:
+            for column in bucket[1].columns:
+                result.append(
+                    (f"{bucket[0]}{NAME_BORDER_SYMBOL}{column}", bucket[1][column])
+                )
+        return result
+
+    @classmethod
+    def _split_data_to_buckets(
+        cls,
+        data: Dataset,
+        compare_by: Literal["groups", "columns", "columns_in_groups", "cross"],
+        target_fields: Union[str, List[str]],
+        baseline_field: Optional[str] = None,
+        group_field: Optional[str] = None,
+    ) -> Tuple:
+        """
+        Splits the given dataset into buckets into baseline and compared data, based on the specified comparison mode.
+
+        Args:
+            data (Dataset): The dataset to be split.
+            group_field (Union[Sequence[str], str]): The field(s) to group the data by.
+            target_fields (Union[str, List[str]]): The field(s) to target for comparison.
+            compare_by (Literal['groups', 'columns', 'columns_in_groups', 'cross'], optional): The method to compare the data. Defaults to 'groups'.
+            baseline_field (Optional[str], optional): The column to use as the baseline for comparison. Required if `compare_by` is 'columns' or 'columns_in_groups'. Defaults to None.
+
+        Returns:
+            Tuple: A tuple containing the baseline data and the compared data.
+
+        Raises:
+            NoRequiredArgumentError: If `baseline_field` is None and `compare_by` is 'columns' or 'columns_in_groups' or 'cross'.
+            ValueError: If `compare_by` is not one of the allowed values.
+        """
+        if compare_by == "groups":
+            if isinstance(target_fields, List) and len (target_fields) > 1:
+                raise ValueError("Only one Target field can be passed when the comparison is done by groups")
+        elif baseline_field is None:
+            raise NoRequiredArgumentError("baseline_field")
+
+        if compare_by != "columns":
+            if group_field is None:
+                raise NoRequiredArgumentError("group_field")
+            elif isinstance(group_field, List):
```

Comment:
what is the reason of this check if u have typing `Optional[str]`?

Reply:
Because it is optional only in csenario of comparison between columns, but not in any other.